src/mono/IosAppBuilderTasks/IosAppBuilderTask.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+public class IosAppBuilderTask : Task
+{
+    /// <summary>
+    /// Path to arm64 AOT cross-compiler (mono-aot-cross)
+    /// It's not used for x64 (Simulator)
+    /// </summary>
+    public string? CrossCompiler { get; set; }
+
+    /// <summary>
+    /// ProjectName is used as an app name, bundleId and xcode project name
+    /// </summary>
+    [Required]
+    public string ProjectName { get; set; } = ""!;
+
+    /// <summary>
+    /// Target directory with *dll and other content to be AOT'd and/or bundled
+    /// </summary>
+    [Required]
+    public string AppDir { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to Mono public headers (*.h)
+    /// </summary>
+    [Required]
+    public string MonoInclude { get; set; } = ""!;
+
+    /// <summary>
+    /// This library will be used as an entry-point (e.g. TestRunner.dll)
+    /// </summary>
+    [Required]
+    public string EntryPointLib { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to a custom main.m with custom UI
+    /// A default one is used if it's not set
+    /// </summary>
+    public string? NativeMainSource { get; set; }
+
+    /// <summary>
+    /// Produce optimized binaries (e.g. use -O2 in AOT)
+    /// </summary>
+    public bool Optimized { get; set; }
+
+    /// <summary>
+    /// Target arch, can be "arm64" (device) or "x64" (simulator) at the moment
+    /// </summary>
+    [Required]
+    public string Arch { get; set; } = ""!;
+
+    /// <summary>
+    /// DEVELOPER_TEAM provisioning, needed for arm64 builds.
+    /// </summary>
+    public string? DevTeamProvisioning { get; set; }
+
+    /// <summary>
+    /// Only generate an xcode project
+    /// </summary>
+    public bool GenerateOnlyXcodeProject { get; set; }
+
+    /// <summary>
+    /// Path to *.app bundle
+    /// </summary>
+    [Output]
+    public string AppBundlePath { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to xcode project
+    /// </summary>
+    [Output]
+    public string XcodeProjectPath { get; set; } = ""!;
+
+    public override bool Execute()
+    {
+        Utils.Logger = Log;
+        bool isDevice = Arch.Equals("arm64", StringComparison.InvariantCultureIgnoreCase);
+        if (isDevice && string.IsNullOrEmpty(CrossCompiler))
+        {
+            throw new ArgumentException("arm64 arch requires CrossCompiler");
+        }
+
+        // escape spaces
+        ProjectName = ProjectName.Replace(" ", "-");
```

Comment:
Why is it needed?

Reply:
It's just a name to be used as an app bundle id (where spaces are not allowed), xcode project name and app name.